app/services/daily_digest_service.rb

```diff
@@ -32,6 +29,43 @@ def execute
 
   private
 
+  def create_updates(updates, user)
+    {
+      community_updates: add_community_updates(user, updates),
+      updates_for_coach: add_updates_for_coach(user)
+
+    }
+  end
+
+  def add_community_updates(user, updates)
+    return [] if user.communities.blank?
+
+    user.communities.pluck(:id).each_with_object({}) do |community_id, updates_for_user|
+      updates_for_user[community_id.to_s] = updates[community_id].dup if updates.include?(community_id)
+    end
+  end
+
+  def add_updates_for_coach(user)
+    coach = user.faculty
+
+    return [] if coach.blank?
+
+    coach.courses.map do |course|
+      pending_submissions = course.timeline_events.pending_review
+      pending_submissions_in_course = pending_submissions.count
+
+      next if pending_submissions_in_course.zero?
```

Comment:
Instead of `next`, this should probably return a `[]` from the block, and `flatten` the result of the block.